### PR TITLE
feat(correlation): use people_url to load people modal

### PR DESCRIPTION
**Pull request by hazzadous**
This change adds a new method to the `personsModelLogic` to allow for
specifying a url to be used to retrieve people from. I've avoided
overloading the existing `loadPeople` and rather added a new one,
`lostPeopleFromUrl`.

Currenlty this is only being used by correlations, but the intention is
to extend it's usage to anywhere else that is performing drilldowns to
people from, for instance, the trends aggregation.

### CODE REVIEW

File: frontend/src/scenes/trends/personsModalLogic.tsx [hunk 1]
```diff
@@ -262,6 +272,20 @@ export const personsModalLogic = kea<personsModalLogicType<PersonModalParams>>({
 
                 return peopleResult
             },
+            loadPeopleFromUrl: async ({ url, stepNumber, breakdown, breakdown_value, label }) => {
+                const people = await (await fetch(url)).json()
```
Comment (EDsCODE) [inline frontend/src/scenes/trends/personsModalLogic.tsx:276]: `await api.get(...)` would be simpler/consistent here

File: frontend/src/scenes/trends/personsModalLogic.tsx [hunk 2]
```diff
@@ -262,6 +272,20 @@ export const personsModalLogic = kea<personsModalLogicType<PersonModalParams>>({
 
                 return peopleResult
             },
+            loadPeopleFromUrl: async ({ url, stepNumber, breakdown, breakdown_value, label }) => {
```
Comment (EDsCODE) [inline frontend/src/scenes/trends/personsModalLogic.tsx:275]: Do you think we could parse some of this directly from the URL itself? That would reduce this function to just requiring a `url`

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 3]
```diff
@@ -1129,19 +1128,42 @@ export const funnelLogic = kea<funnelLogicType>({
                 funnelStep: stepNumber,
             })
         },
-        openCorrelationPersonsModal: ({ entity, converted, resultType }) => {
-            personsModalLogic.actions.loadPeople({
-                action: { id: entity.id, name: entity.name, properties: entity.properties, type: entity.type },
-                label: entity.id,
-                date_from: '',
-                date_to: '',
-                filters: {
```
Comment (EDsCODE) [inline frontend/src/scenes/funnels/funnelLogic.ts:1138]: great to see all this removed

**Review by EDsCODE — APPROVED** [review]
Some questions on making some bits even simpler. Otherwise lgtm

File: frontend/src/scenes/trends/personsModalLogic.tsx (hunk 2 above)
Comment (hazzadous) [reply frontend/src/scenes/trends/personsModalLogic.tsx:275]: Yeah was thinking about that. These additional args are specific things for generating the label for display. I wouldn't want to parse the url, if smells. I don't want to start coupling things like this to what can be, at the moment, be treated as an opaque url.

If we really care about making sure that the label consistent with the data returned, we can either pass along an object along with the url, containing the info (could be the exact label although that seems a little prescriptive for what feels contextual to the consumer). Or we can return this data within the response?

**Comment by hazzadous** [comment]
Just merging in master to see if it prompts the e2e tests to pass

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 4]
```diff
@@ -1129,19 +1128,42 @@ export const funnelLogic = kea<funnelLogicType>({
...
-            })
-            eventUsageLogic.actions.reportCorrelationInteraction(resultType, 'person modal', { ...entity, converted })
+        openCorrelationPersonsModal: ({ correlation, success }) => {
+            if (correlation.result_type === FunnelCorrelationResultsType.Properties) {
+                const { breakdown, breakdown_value } = parseBreakdownValue(correlation.event.event)
+
+                personsModalLogic.actions.loadPeopleFromUrl({
+                    url: success ? correlation.success_people_url : correlation.failure_people_url,
+                    // I'm not sure why this is -2, I'm just copying from what was here previously
+                    stepNumber: success ? values.stepsWithCount.length : -2,
```
Comment (neilkakkar) [inline frontend/src/scenes/funnels/funnelLogic.ts:1138]: don't need this anymore, it's already in the url now? (was https://github.com/PostHog/posthog/blob/master/frontend/src/scenes/funnels/funnelLogic.ts#L1129 funnelStep, which we've overridden with `funnel_custom_steps` .

File: frontend/src/scenes/funnels/funnelLogic.ts [hunk 5]
```diff
@@ -1129,19 +1128,42 @@ export const funnelLogic = kea<funnelLogicType>({
...
+                    values.filters.funnel_correlation_person_entity
+                )
+            } else {
+                const { name, properties } = parseEventAndProperty(correlation.event)
+
+                personsModalLogic.actions.loadPeopleFromUrl({
+                    url: success ? correlation.success_people_url : correlation.failure_people_url,
+                    stepNumber: success ? values.stepsWithCount.length : -2,
+                    label: name,
+                    breakdown: '',
```
Comment (neilkakkar) [inline frontend/src/scenes/funnels/funnelLogic.ts:1156]: StepNumber, and breakdown don't make sense here anymore?

File: frontend/src/scenes/funnels/funnelLogic.ts (hunk 4 above)
Comment (hazzadous) [reply frontend/src/scenes/funnels/funnelLogic.ts:1138]: These additional vars I kept as I thought they are needed for setting the title of the modal. I'll remove if not @neilkakkar 


If they are still needed, I'd rather keep this PR small. We could follow up with an addition to the people endpoint to include the specific filtering detail that could then be used for display, or have this to hand in an easy way in the initial response?

I'll add a NOTE about why it's still needed, unless you think we can remove them @neilkakkar

File: frontend/src/scenes/funnels/funnelLogic.ts (hunk 5 above)
Comment (hazzadous) [reply frontend/src/scenes/funnels/funnelLogic.ts:1156]: I think stepNumber is still needed to have the right modal title, although I'm just about to test this.

The breakdown and breakdown_value yup I guess shouldn't need to set that here, I'll make them optional.

File: frontend/src/scenes/funnels/funnelLogic.ts (hunk 4 above)
Comment (neilkakkar) [reply frontend/src/scenes/funnels/funnelLogic.ts:1138]: Ah right, this tells whether it's success or failure? Cool cool

File: frontend/src/scenes/trends/personsModalLogic.tsx (hunk 1 above)
Comment (hazzadous) [reply frontend/src/scenes/trends/personsModalLogic.tsx:281]: Have updated

File: frontend/src/scenes/funnels/funnelLogic.ts (hunk 5 above)
Comment (hazzadous) [reply frontend/src/scenes/funnels/funnelLogic.ts]: Simplified a little, but yeah, need this for display.

Can clean up in another PR

**Pull request merged**